### ISSUE TITLE
Need to manually refresh file explorer in WSL to see changes, both windows10 and windows11

### ISSUE THREAD

**Issue by neeeeecka** [microsoft/WSL#7941]
### Version

Microsoft Windows [Version 10.0.22000.434]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.60.1

### Distro Version

Any distro 

### Repro Steps

1. Open wsl folder from file explorer on win11 or from `explorer.exe .` command. Open one more so you have 2 windows open.
2. Create/delete/rename a file from one window or from terminal
3. To see changes in folder/s you need to manually Right click -> more options -> refresh(On win11) or Right click -> refresh(on win10)

### Expected Behavior

Files should update automatically in the explorer windows without having to manually refresh

### Actual Behavior

You have to manually refresh explorer window/s to see the file changes

**Comment by equation-group** [comment]
This is the expected behaviour,  wsl folder is not native windows folder and being used by linux.
you can close this issue.

**Comment by nook24** [comment]
This effects me as well and honestly makes Files pretty unusable with WSL. Would be good if this could be fixed

**Comment by codeananda** [comment]
Ahh I see I've been duped... *starts reading article explaining how to dual boot*

**Comment by iwiwdt** [comment]
> WSL is snake oil, i suggest you to dualboot or use docker :)

man, that's wild. Do you also consider hypervisors and virtualization to be snake oil? I'm guessing you'd also say "Java is slow because it runs in a VM" or something like that, wouldn't you?

**Comment by DryreL** [comment]
+1

**Comment by amalizzio** [comment]
+1 please add this

**Comment by barbuslex** [comment]
+1

**Comment by PolarGoose** [comment]
This issue is still relevant.
Has there been any workaround for that?

**Comment by codeananda** [comment]
Update: I switched to Ubuntu as my main operating system and haven’t looked back.

**Comment by PolarGoose** [comment]
> Update: I switched to Ubuntu as my main operating system and haven’t looked back.

Now, with the WSL2 GUI passthrough, you can run GUI apps like VsCode or a Ubuntu file manager from Linux itself, and they will not have problems with file updates inside the Linux file system.

The root cause of this problem is that the Linux file system is exposed to Windows via SAMBA or a similar network file-sharing mechanism that doesn't support file update notifications. Therefore, Windows is completely blind to any file updates inside the Linux file system.
The same problem is with VMware and VirtualBox built-in shared drive functionality.

**Comment by nook24** [comment]
> The root cause of this problem is that the Linux file system is exposed to Windows via SAMBA or a similar network file-sharing mechanism that doesn't support file update notifications. Therefore, Windows is completely blind to any file updates inside the Linux file system.
The same problem is with VMware and VirtualBox built-in shared drive functionality.

I just want to mention that is is not entirely correct. Windows File Explorer can and does monitor for changes on a SAMBA share. You can test this by open up a SAMBA share in Windows Explorer, than create a new file `touch test.txt` and it will show up in Explorer within a second or so.
Same for removing the file. 

But i cannot speak for VirtualBox as i have not tested this. Looks like Microsoft is not using SAMBA under the hood, but as you said a 
> similar network file-sharing mechanism that doesn't support file update notifications

**Comment by PolarGoose** [comment]
> I just want to mention that is is not entirely correct. Windows File Explorer can and does monitor for changes on a SAMBA share.

Yes, sorry for the confusion. In the wiki, they explicitly mention that they support this scenario: [SAMBA wiki - File_Change_Notification](https://wiki.samba.org/index.php/Editing_files_locally_on_server:_interoperability#File_Change_Notification).

**Comment by warpdesign** [comment]
I don't understand: I received a notification that says "Closed this as completed". But here I read it has been closed for.. inactivity?

Funny the Microsoft bot automatically closes stalled issues as "completed" :)
